packages/contracts/tasks/deploy.ts

```diff
@@ -1,60 +1,92 @@
       QuadraticVoteCollectModule: [
-        "0x60Ae865ee4C725cd04353b5AAb364553f56ceF82",
-        "0x1353aAdfE5FeD85382826757A95DE908bd21C4f9",
+        lensPolygonAddresses.lensHubImplementation,
+        lensPolygonAddresses.moduleGlobals,
       ],
     };
 
+    const constructorArgumentsMumbai: Record<Contracts, string[]> = {
+      QuadraticFundingCurator: ['0x45cf9Ba12b43F6c8B7148E06A6f84c5B9ad3Dd44', admin.address],
+      QuadraticVoteCollectModule: [
+        lensMumbaiAddresses.lensHubImplementation,
+        lensMumbaiAddresses.moduleGlobals,
+      ],
+    };
+
+   const toFile = (path: string, deployment: Record<Contracts, string>) => {
+      fs.writeFileSync(path, JSON.stringify(deployment), { encoding: 'utf-8' });
+    }
+
     for (const [name, contract] of Object.entries(contracts)) {
       console.log(`Starting deployment of ${name}`);
       const factory = contract;
-      const constructorArgs = Object.entries(constructorArguments).find((entry) => entry[0] === name)?.[1];
-      console.log(`Constructor arguments: ${constructorArgs}`);
+      let constructorArgs;
+
+      if (hre.network.name == 'polygon-mainnet') {
+        constructorArgs = Object.entries(constructorArguments).find((entry) => entry[0] === name)?.[1];
+        console.log(`Constructor arguments: ${constructorArgs}`);
+      } else if (hre.network.name == 'polygon-mumbai') {
+        constructorArgs = Object.entries(constructorArgumentsMumbai).find((entry) => entry[0] === name)?.[1];
+        console.log(`Constructor arguments: ${constructorArgs}`);
+      } else if (hre.network.name == 'hardhat') {
+       constructorArgs = Object.entries(constructorArgumentsMumbai).find((entry) => entry[0] === name)?.[1];
+       console.log(`Constructor arguments: ${constructorArgs}`);
+      } else {
+          throw new Error('Not a supported network');
+      }
 
       const instance = constructorArgs ? await factory.deploy(...constructorArgs) : await factory.deploy();
+      
       await instance.deployed();
 
       console.log(`${name} is deployed to address: ${instance.address}`);
+      
       deployments[name as Contracts] = instance.address;
-
-      if (hre.network.name !== "hardhat") {
+    
+      toFile(`deployments/deployments-${hre.network.name}.json`, deployments);
+  
+      if (hre.network.config.chainId !== 31337) {
         try {
           const code = await instance.instance?.provider.getCode(instance.address);
-          if (code === "0x") {
+          if (code === '0x') {
             console.log(`${instance.name} contract deployment has not completed. waiting to verify...`);
             await instance.instance?.deployed();
           }
-          await hre.run("verify:verify", {
+
+          await hre.run('verify:verify', {
             address: instance.address,
-            constructorArgs,
+            contract: `contracts/${name}.sol:${name}`,
```

Comment:
This contract should already be available in this script. Why are you loading from a file?

Reply:
this is where it broke on the cache issue.  loading the specific contract fixed the caching error.

Reply:
Removed it to find out what happens. This is not needed in any other project I set up with these scripts.